### PR TITLE
Added chapter about live-preview

### PR DESCRIPTION
**Pull request by wachterjohannes**
This chapter extend the cookbook with a entry for the live preview syntax.

### CODE REVIEW

File: cookbook/live-preview.rst [hunk 1]
```diff
@@ -0,0 +1,171 @@
+How to include live-preview in my template?
+===========================================
+
+Sulu provides a powerfull live-preview system which can be used by every website
```
Comment (thomasduenser) [inline cookbook/live-preview.rst]: @wachterjohannes `powerful`

File: cookbook/live-preview.rst [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        </properties>
+    </template>
+
+This page consists of a single title and a url. The template is quite basic but
+includes the RDFa attribute `property=title` to tell sulu that the title is
+rendered in this container.
+
+When the title of the page will be updated in the content form the admin will
+send this change to the server where the twig block `content` will be rendered.
+The server response only the part which was changed (indicated by
```
Comment (thomasduenser) [inline cookbook/live-preview.rst]: I would write `The server returns only the part ...` or `The server response contains only the part ...` but it sounds quite confusing right now.

File: cookbook/live-preview.rst [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+`property=title`). Back in the javascript application the preview on the right
+will update the container which is marked with the particular property.
+
+This process ensures a quite good performance (as good as the rendering on the
+server side) and minimize the load of the requests.
```
Comment (thomasduenser) [inline cookbook/live-preview.rst]: @wachterjohannes I think it should be `... and minimizes ...`

File: cookbook/live-preview.rst [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+.. code-block:: twig
+
+    <a href="{{ sulu_content_path(content.link.url) }}" property="link">
+        {{ content.link.title }}
+    </a>
+
+Images
+******
+
+As like the links you can simple use the src-attribute to update an image
```
Comment (thomasduenser) [inline cookbook/live-preview.rst]: @wachterjohannes I would rewrite this sentence. Maybe: `To update an image you can simply use the src-attribute when the mentioned property has changed`

File: cookbook/live-preview.rst [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+.. code-block:: twig
+
+    <img src="{{ image.thumbnails['170x170'] }}" alt="{{ image.title }}"/>
+
+
+Multiple values
+***************
+
+For multiple values simple use table, ul, ol or div tags and the content will
```
Comment (thomasduenser) [inline cookbook/live-preview.rst]: @wachterjohannes `simply`

File: cookbook/live-preview.rst [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+        {% for category in content.categories %}
+            <li>{{ category.name }}</li>
+        {% endfor %}
+    </ul>
+
+Snippets
+********
+
+For the snippet content-type all the selected snippets has to be updated.
+Therefor you only have to set the upper property.
```
Comment (thomasduenser) [inline cookbook/live-preview.rst]: @wachterjohannes `Therefore`

File: cookbook/live-preview.rst [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+                    <property name="url" type="resource_locator" mandatory="true">
+                        <tag name="sulu.rlp"/>
+                    </property>
+                </properties>
+            </section>
+        </properties>
+    </template>
+
+This page consists of a single title and a url. The template is quite basic but
+includes the RDFa attribute `property=title` to tell sulu that the title is
```
Comment (danrot) [inline cookbook/live-preview.rst]: `property="title"`

File: cookbook/live-preview.rst [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+rendered in this container.
+
+When the title of the page will be updated in the content form the admin will
+send this change to the server where the twig block `content` will be rendered.
+The server response contains only the part which was changed (indicated by
+`property=title`). Back in the javascript application the preview on the right
```
Comment (danrot) [inline cookbook/live-preview.rst]: `property="title"`

File: cookbook/live-preview.rst [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+        {% for block in content.blocks %}
+            <div rel="blocks" typeof="block">
+                <div property="title">{{ block.title }}</div>
+            </div>
+        {% endfor %}
+    </div>
+
+You have to define the `property="blocks"` as `typeof="collection"` and each
+item of the block as `typeof="block"` and set the relation to the parent
+property in this case "blocks".
```
Comment (danrot) [inline cookbook/live-preview.rst]: `property, in this case "blocks"`

File: cookbook/live-preview.rst [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+
+With this definitions the system is able to update only the `title` of the
```
Comment (danrot) [inline cookbook/live-preview.rst]: `With these definitions`

File: cookbook/live-preview.rst [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+first block item and don't response the whole div-container of the
```
Comment (danrot) [inline cookbook/live-preview.rst]: `and doesn't have to return the entire container of the block-property in the response`

**Pull request merged**